app/controllers/concerns/collection.rb

```diff
@@ -23,5 +23,9 @@ def sort(data)
 
       data.order(build_order_by(data.klass, params[:sort_by]))
     end
+
+    def sorted?(data)
+      data.to_sql.include?('ORDER')
```

Comment:
I haven't found a better way of detecting if the dataset is sorted or not, but if there's any, please let me know. Another alternative would be a default scope, but that doesn't really work well with `sort_by` and it's also not recommended for use for anything more complex than ASC/DESC.

Reply:
I guess the `.order` call writes the state somewhere. Try if that can be read. Otherwise only arel comes to mi mind, traversing the tree checking for an ORDER node.

Reply:
Wouldn't it be better to move the check to the controller level? Basically checking whether the `sort_by` was used or not. There might be a default sort set somewhere in the upper levels which could impact the default sort of the controller.

Reply:
I'd implement default sort in the same way we implement [default search in some controllers](https://github.com/RedHatInsights/compliance-backend/blob/master/app/controllers/v1/profiles_controller.rb#L16).

Reply:
@akofink we can't really do that here as the `score` is not a field, but a dynamic value generated from a method. There might be a way though to fetch that value via some querying magic, will look into that...

Reply:
Hmm, good point. I'm now questioning why we sort by score in the first place :laughing: Since it cannot be sorted by from the user, it doesn't make sense to use it by default, since we'd have to show it in the API response meta section. I went digging, and it doesn't seem to have been added with any particular use case in mind: https://github.com/RedHatInsights/compliance-backend/pull/35/files#diff-7a47d335107fc35c2e79cc9c2b0a195c863594dd2343bcd48a4c770ac4af673cR9 I wonder if we should remove it?

Reply:
I fully support this idea as it would save me from a lot of SQL thinking :exploding_head: